### PR TITLE
[WooExpress removal] Login code cleanup

### PR DESCRIPTION
**Pull request by hichamboushaba**
Part of: #11115 

### Description
This PR removes the unused code of store creation, sorry for the size of the PR, but most of the code is just code removal, and should be easy to review.
The only commits which contained some changes are the following:
1. Update the theme's installation logic to remove all bits related to store creation in ea685c0c70a30bc45d163d4a1824ac1e7b2fb034
2. Moved different classes that were under the package `storecreation` to other more appropriate packages in 7d798914ec9234c665939dce91bff654bade84a3

I'll handle remaining code cleanup on separate PRs.

### Testing instructions
Smoke test the following parts:
1. Login.
2. Theme installation.
3. Domains.

### Images/gif

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/iap/CheckIapEligibilityFragment.kt [hunk 1]
```diff
@@ -57,11 +56,6 @@ class CheckIapEligibilityFragment : BaseFragment() {
     }
 
     private fun navigateToCreateStoreSummaryStep() {
-        findNavController()
-            .navigateSafely(
-                CheckIapEligibilityFragmentDirections
-                    .actionCheckIapEligibilityFragmentToStoreCreationSummaryFragment(),
-                skipThrottling = true
-            )
+        TODO()
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/iap/CheckIapEligibilityFragment.kt:59]: This is just temporary, on a next PR, I'll move all IAP code to `debug`, and I'll remove this fragment.

**Review by 0nko — APPROVED** [review]
Nice job, Hicham! 👍  Everything's looking and working well. I just noticed that there was still `com.woocommerce.android.ui.login.storecreation`. Were you planning to remove it in another PR?

**Comment by hichamboushaba** [comment]
> I just noticed that there was still com.woocommerce.android.ui.login.storecreation. Were you planning to remove it in another PR?

Thanks @0nko, yes, it still has some IAP and plan upgrades code that I'm planning to remove on other PRs.

**Pull request merged**